### PR TITLE
mySQL_database

### PR DESCRIPTION
**Pull request by koval1off**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def get_number_from_input(text: str) -> int:
 
 
 def validation_credential(user_id: str, user_pin: str) -> bool:
-    if len(str(user_id)) == 4 and len(str(user_pin)) == 4:
+    if len(user_id) == 4 and len(user_pin) == 4:
```
Comment (koval1off) [inline main.py:16]: I don't kwon why I previously changed the datatype here. I guess I just got confused then

File: bank.py [hunk 2]
```diff
@@ -86,11 +96,17 @@ def change_user_pin(self, user: User, new_pin: str) -> None:
...
-
-    def save_users(self) -> None:
-        """loads all information about users in JSON"""
-        json_string = json.dumps([user.__dict__ for user in self.users])
-
-        with open("database.json", "w") as file:
-            file.write(json_string)
+        self.save_users(user)
+
+    def save_users(self, user: User) -> None:
```
Comment (koval1off) [inline bank.py]: I don't really know if it's OK connecting to database several times. Maybe I must to do it only once?

File: main.py (hunk 1 above)
Comment (Alex-Krykun) [reply main.py:16]: good point!

File: bank.py [hunk 3]
```diff
@@ -1,22 +1,32 @@
...
-            data_users = json.load(file)
-            for data in data_users:
-                user = User(data["user_id"],
-                            data["pin"],
-                            data["balance"])
-                users.append(user)
+        mydb = mysql.connector.connect(
+            host="localhost",
+            user="root",
+            password="Kovalsql#12345",
```
Comment (Alex-Krykun) [inline bank.py]: remove real password and leave empty

Comment (Alex-Krykun) [reply bank.py]: remove real user and leave empty blank

File: bank.py (hunk 2 above)
Comment (Alex-Krykun) [reply bank.py]: rename it to save_user or update_user

Comment (Alex-Krykun) [reply bank.py]: you need to connect each time when operating with it
if you will keep connection open is bad for performance

**Pull request merged**